### PR TITLE
Class datarule migration

### PR DESCRIPTION
**Pull request by dhuebner**

### CODE REVIEW

File: com.regnosys.rosetta.lib/src/main/java/model/annotations.rosetta [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace "com.rosetta.model"
+version "${project.version}"
+
+anno keyed:
```
Comment (jim-h-wang) [inline com.regnosys.rosetta.lib/src/main/java/model/annotations.rosetta]: Can we change the keyword to `annotation`?

File: com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext [hunk 2]
```diff
@@ -19,27 +19,35 @@ RootElement:
...
-	'data' RosettaNamed ':' Definable?
-	Annotations?
+	'data' RosettaNamed ('extends' superType = [Data|ValidID])? ':' Definable?
+	(Annotations| ClassSynonyms)*
 	attributes += Attribute+
 	conditions += Condition*
 ;
 
+Attribute:
+	RosettaNamed RosettaTyped card=RosettaCardinality Definable? (Annotations|Synonyms)*;
```
Comment (jim-h-wang) [inline com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext]:
```suggestion
	RosettaNamed RosettaTyped card=RosettaCardinality Definable? (Annotations|Synonyms)*
```

File: com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext [hunk 3]
```diff
@@ -51,35 +59,33 @@ Function:
...
-;
-
 PostCondition returns Condition: // TODO no name? annotations?
 	'post-condition' ':' Definable?
 	(expressions+=RosettaCalcExpression ';')+
 ;
 
+Operation: // TODO no name? annotations?
+	'assign-output' attribute = [Attribute|ValidID] ':' Definable?
+	expression = RosettaCalcExpression ';'?
```
Comment (jim-h-wang) [inline com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext]: if we don't need the `;` then let's avoid them.
```suggestion
	expression = RosettaCalcExpression
```

File: com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext [hunk 4]
```diff
@@ -51,35 +59,33 @@ Function:
...
 	'alias' RosettaNamed ':' Definable?  expression=RosettaCalcExpression
 ;
 
-Attribute:
-	RosettaNamed RosettaTyped card=RosettaCardinality Definable? Annotations?;
 
 Condition:
 	'condition' RosettaNamed? ':' Definable?
 		Annotations?
 		(expressions+=RosettaCalcExpression ';')+
```
Comment (jim-h-wang) [inline com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext]: Is there any way we can remove the need for the `;` here?

Comment (dhuebner) [reply com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext]: Not with ´RosettaCalcExpression´ following by ´RosettaCalcExpression´. For `condition` I was thinking about having only one expression.

File: com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext (hunk 3 above)
Comment (dhuebner) [reply com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext]: Okay

File: com.regnosys.rosetta.lib/src/main/java/model/annotations.rosetta (hunk 1 above)
Comment (dhuebner) [reply com.regnosys.rosetta.lib/src/main/java/model/annotations.rosetta]: Sure

**Pull request merged**